homeassistant/components/picnic/services.py

```diff
@@ -0,0 +1,131 @@
+"""Services for the Picnic integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from python_picnic_api import PicnicAPI
+import voluptuous as vol
+
+from homeassistant.core import HomeAssistant, ServiceCall
+from homeassistant.helpers import device_registry
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    ATTR_AMOUNT,
+    ATTR_DEVICE_ID,
+    ATTR_PRODUCT_ID,
+    ATTR_PRODUCT_IDENTIFIERS,
+    ATTR_PRODUCT_NAME,
+    CONF_API,
+    DOMAIN,
+    SERVICE_ADD_PRODUCT_TO_CART,
+)
+
+
+class PicnicServiceException(Exception):
+    """Exception for Picnic services."""
+
+
+async def async_register_services(hass: HomeAssistant) -> None:
+    """Register services for the Picnic integration, if not registered yet."""
+
+    # Register the add_product service
+    if not hass.services.has_service(DOMAIN, SERVICE_ADD_PRODUCT_TO_CART):
```

Comment:
Preferring an early return here can let the rest of this function get un-indented